### PR TITLE
FLUID-6548: Update linting for Infusion based on new rules and linter

### PR DESCRIPTION
**Pull request by jobara**
Includes an update to use fluid-lint-all for linting.

https://issues.fluidproject.org/browse/FLUID-6548

### CODE REVIEW

File: src/framework/core/js/Fluid.js [hunk 1]
```diff
@@ -205,27 +203,26 @@ var fluid = fluid || fluid_3_0_0;
...
         var activity = fluid.makeArray(fluid.describeActivity()); // Take copy since we will destructively modify
         fluid.popActivity(activity.length); // clear any current activity - TODO: the framework currently has no exception handlers, although it will in time
         if (fluid.failureEvent) { // notify any framework failure prior to successfully setting up the failure event below
-            fluid.failureEvent.fire(args, activity);
+            fluid.failureEvent.fire(messages, activity);
         } else {
-            fluid.logFailure(args, activity);
-            fluid.builtinFail(args, activity);
+            fluid.logFailure(messages, activity);
+            fluid.builtinFail(messages, activity);
```
Comment (jobara) [inline src/framework/core/js/Fluid.js:218]: @amb26 I want to highlight this change to make sure you see it. There is another similar one elsewhere in the code as well. The change here is to use the [rest parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters) in stead of arguments. The rest parameter returns a named array of the arguments from that position.

File: src/framework/core/js/Fluid.js [hunk 2]
```diff
@@ -2321,26 +2364,27 @@ var fluid = fluid || fluid_3_0_0;
...
      * fine-grained policy. If it is an Object, the values may also themselves be EL paths
      * representing that a default value is to be taken from that path.
-     * @param {...Object} options1, options2, .... - an arbitrary list of options structure which are to
+     * @param {...Object} sources - an arbitrary list of options structure which are to
      * be merged together. These will not be modified.
      */
 
-    fluid.merge = function (policy /*, ... sources */) {
-        var sources = Array.prototype.slice.call(arguments, 1);
+    fluid.merge = function (policy , ...sources) {
```
Comment (jobara) [inline src/framework/core/js/Fluid.js:2387]: @amb26 this is another case where I've converted to use a [rest parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters). In this case the other benefit is that the first parameter called `policy` is not included in the `sources` array, so we don't need to handle skipping it later in the code.

File: src/framework/core/js/Fluid.js (hunk 1 above)
Comment (amb26) [reply src/framework/core/js/Fluid.js:218]: I assume that the use of the spread pattern makes the use of fluid.makeArray() unnecessary on the first line

Comment (jobara) [reply src/framework/core/js/Fluid.js:218]: @amb26 yes, the call to `fluid.makeArray` that had been on line 214 was removed for this reason. The call to `fluid.makeArray` on line 212 (formerly 215) seems be used for making a copy.

**Comment by the-t-in-rtf** [comment]
The pull against fluid-grunt-lint-all associated with this has been merged and I have cut the 2.0.0 release.   Please update this pull to use the released version, and then we should be good to merge.

**Comment by greatislander** [comment]
@jobara This one needs an update as well.

**Comment by jobara** [comment]
> @jobara This one needs an update as well.

Was this in reference to a specific dependency or just about updating from "main"?

**Comment by greatislander** [comment]
Updating from `main`.

**Comment by greatislander** [comment]
The changes in [FLUID-6573](https://issues.fluidproject.org/browse/FLUID-6573) ([pull request](https://github.com/fluid-project/infusion/pull/1032)) should get the build working again here.

**Comment by jobara** [comment]
The Codecov check for this PR is interesting. It seems the patch is only getting 72.72% for coverage, but for the project the coverage actually increased.

**Comment by greatislander** [comment]
> The Codecov check for this PR is interesting. It seems the patch is only getting 72.72% for coverage, but for the project the coverage actually increased.

So I guess the overall lines of code decreased and thus the project coverage is higher overall, but the changed lines only have 72.72% coverage so that check is failing?

**Comment by amb26** [comment]
I found that our current coverage thresholds are a little aggressive - my pull https://github.com/fluid-project/infusion/pull/972 similarly had the patch miss by 0.09% and was so thrown out ... even though, again, coverage improved overall. It might be an idea to replace the test with a simple check that coverage has not reduced overall or at least below 88%

**Comment by greatislander** [comment]
We can disable the patch check if desired (or simply lower the threshold for patches) — see docs: https://docs.codecov.io/docs/commit-status#branches

**Comment by jobara** [comment]
> We can disable the patch check if desired (or simply lower the threshold for patches) — see docs: https://docs.codecov.io/docs/commit-status#branches

That's an option. @amb26, @the-t-in-rtf, @cindyli any thoughts?

**Comment by amb26** [comment]
I think eliminating the patch check is my favoured option - there are lots of reasons why a patch in itself may not appear to have adequate coverage (e.g. it consists mostly of deletions of uncovered code!)

**Comment by jobara** [comment]
> I think eliminating the patch check is my favoured option - there are lots of reasons why a patch in itself may not appear to have adequate coverage (e.g. it consists mostly of deletions of uncovered code!)

I'm disabling the patch checks in this PR.

**Comment by jobara** [comment]
@amb26 this PR is ready for review.

**Pull request merged**